### DISCUSSION TITLE
sendAsync(data, callback) is deprecated, please use window.ethereum.request(data) instead.

### DISCUSSION THREAD

**Discussion by 0xBeycan** [trustwallet/wallet-core#3022] [Q&A]
**Describe the bug**
When I try to use it with Web3JS, I get the error in the title. While all other vallets support the code I am using, even the Mobile TrustWallet via WalletConnect is running. Strangely, the Chrome extension is not working.

I include the TrustWallet ethereum provider in Web3 in the Chrome extension. Later, when I use methods such as balance, blockNumber with Web3, and create a contract instance and try to get token information, I get the error in this title.

**Comment by george-gkas** [comment]
Hello @BeycanDeveloper, thanks for reaching out.

We have dedicated developer docs to walk you through the process with examples and answer your questions. Please look at [Trust Developer](https://developer.trustwallet.com/developer/develop-for-trust/browser-extension/evm) docs and let us know if you still have issues. 

We would also appreciate it if you could provide us with the code snippet you have tried and the contract address you are trying to interact with so we can test it internally.
